### PR TITLE
[API][MyTeam] Add and modify my team member

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: features/api/my_teams.feature [hunk 1]
```diff
@@ -52,3 +54,173 @@ Feature:
     }
     """
 
+  Scenario: As a referent I cann't add a new member in my team with invalid data
```
Comment (elvino29) [inline features/api/my_teams.feature:57]:
```suggestion
  Scenario: As a referent I cannot add a new member in my team with invalid data
```

File: features/api/my_teams.feature [hunk 2]
```diff
@@ -52,3 +54,173 @@ Feature:
...
+                "message": "Le militant choisi ne fait pas partie de la zone géographique que vous gérez.",
+                "propertyPath": "adherent"
+            },
+            {
+                "propertyPath": "role",
+                "message": "Cette position n'est pas valide."
+            },
+            {
+                "propertyPath": "scope_features",
+                "message": "Une ou plusieurs des accès délégués ne sont pas valides."
```
Comment (elvino29) [inline features/api/my_teams.feature:94]: missing word here

File: features/api/my_teams.feature [hunk 3]
```diff
@@ -52,3 +54,173 @@ Feature:
...
+                "propertyPath": "adherent",
+                "message": "Ce militant n'est pas disponible pour l'ajout dans l'équipe."
+            },
+            {
+                "message": "Le militant choisi ne fait pas partie de la zone géographique que vous gérez.",
+                "propertyPath": "adherent"
+            },
+            {
+                "propertyPath": "role",
+                "message": "Cette position n'est pas valide."
```
Comment (elvino29) [inline features/api/my_teams.feature:90]: position or role?

File: features/api/my_teams.feature [hunk 4]
```diff
@@ -52,3 +54,173 @@ Feature:
...
+            },
+            {
+                "propertyPath": "scope_features",
+                "message": "Une ou plusieurs des accès délégués ne sont pas valides."
+            }
+        ]
+    }
+    """
+
+  Scenario: As a referent I cann't add a new member in my team, if the user is already in my team
```
Comment (elvino29) [inline features/api/my_teams.feature:100]:
```suggestion
  Scenario: As a referent I cannot add a new member in my team, if the user is already in my team
```

File: features/api/my_teams.feature [hunk 5]
```diff
@@ -52,3 +54,173 @@ Feature:
...
+        "violations": [
+            {
+                "propertyPath": "adherent",
+                "message": "Le militant fait déjà partie de cette équipe."
+            }
+        ]
+    }
+    """
+
+  Scenario: As a referent I cann't add a new member in my team, if the user is not in my managed zone
```
Comment (elvino29) [inline features/api/my_teams.feature:131]:
```suggestion
  Scenario: As a referent I cannot add a new member in my team, if the user is not in my managed zone
```

File: src/Entity/MyTeam/Member.php [hunk 6]
```diff
@@ -18,6 +21,32 @@
  * })
  *
  * @UniqueEntity(fields={"team", "adherent"}, errorPath="adherent", message="my_team.member.adherent.already_in_collection")
+ *
+ * @ApiResource(
+ *     shortName="MyTeamMember",
+ *     attributes={
+ *         "normalization_context": {
+ *             "iri": true,
```
Comment (elvino29) [inline src/Entity/MyTeam/Member.php:29]: why?

File: src/Entity/MyTeam/Member.php [hunk 7]
```diff
@@ -18,6 +21,32 @@
...
+ *     itemOperations={
+ *         "put": {
+ *             "path": "/v3/my_team_members/{id}",
+ *             "requirements": {"id": "%pattern_uuid%"},
+ *         },
+ *     },
+ *     collectionOperations={
+ *         "post": {
+ *             "path": "/v3/my_team_members",
+ *             "denormalization_context": {"groups": {"my_team_member_write", "my_team_member_post"}}
```
Comment (elvino29) [inline src/Entity/MyTeam/Member.php:46]: `my_team_member_post` seems never used

File: src/Validator/MyTeamMember.php [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * @Annotation
+ * @Target({"PROPERTY", "ANNOTATION"})
+ */
+class MyTeamMember extends Constraint
+{
+    public $messageInvalidAdherentSource = "Ce militant n'est pas disponible pour l'ajout dans l'équipe.";
```
Comment (elvino29) [inline src/Validator/MyTeamMember.php:13]:
```suggestion
    public $messageInvalidAdherentSource = "Ce militant ne peut pas être ajouter à l'équipe.";
```

File: translations/validators.fr.yml [hunk 9]
```diff
@@ -571,8 +571,8 @@ scope.zone.local_scope_with_unmanaged_zone: La zone spécifiée n'est pas géré
 #
 # MyTeam
 #
-my_team.member.adherent.already_in_collection: L'adhérent "%value%" fait déjà partie de cette équipe.
+my_team.member.adherent.already_in_collection: Le militant fait déjà partie de cette équipe.
 my_team.member.adherent.not_blank: Veuillez sélectionner un militant.
 my_team.member.role.not_blank: Veuillez sélectionner une position.
 my_team.member.role.invalid_choice: Cette position n'est pas valide.
-my_team.member.scope_features.invalid_choice: Accès délégués ne sont pas valides.
+my_team.member.scope_features.invalid_choice: Une ou plusieurs des accès délégués ne sont pas valides.
```
Comment (elvino29) [inline translations/validators.fr.yml:578]: missing word here

File: features/api/my_teams.feature (hunk 3 above)
Comment (illusionOfParadise) [reply features/api/my_teams.feature:90]: In french it's `position`.

File: features/api/my_teams.feature (hunk 2 above)
Comment (illusionOfParadise) [reply features/api/my_teams.feature:94]: Which one?

File: src/Entity/MyTeam/Member.php (hunk 7 above)
Comment (illusionOfParadise) [reply src/Entity/MyTeam/Member.php:46]: Here https://github.com/EnMarche/en-marche.fr/pull/6796/files#diff-f9849126c0efaf10533e7b570a27c63274b924b07c429fe2263d3c42ea2b58fcR72

File: translations/validators.fr.yml (hunk 9 above)
Comment (illusionOfParadise) [reply translations/validators.fr.yml:578]: Which one?

Comment (elvino29) [reply translations/validators.fr.yml:578]: sorry I misread

File: features/api/my_teams.feature (hunk 2 above)
Comment (elvino29) [reply features/api/my_teams.feature:94]: sorry I misread

**Review by elvino29 — APPROVED** [review]

**Pull request merged**